### PR TITLE
fix(slider): Fix style change for updated slider and directionality change

### LINKED ISSUE #6542: [slider] change detection issue with styles
#### Bug, feature request, or proposal:
Bug

#### Current

Note that the slider track does not update its position until mouseover

slider-cd-issue

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Latest build:  0113c38

### PR DESCRIPTION
**Pull request by g1shin**
1. Before the fix: the slider style was not getting updated upon blur:  
slider-cd-issue
Fixes #6542 

2. Fixes directionality change not re-rendering updated slider.

### CODE REVIEW

**Comment by mmalerba** [comment]
I'm not sure this is the right fix, doesn't make sense to blur it on click

**Comment by g1shin** [comment]
Changes have been made to this PR. Ready for review now :+1:

**Comment by g1shin** [comment]
Changes made based on discussion. Ready for review again

**Comment by g1shin** [comment]
I added a change to fix prerender test.

**Comment by g1shin** [comment]
Moved the subscription to directionality change into `ngOnInit()` as well

**Comment by g1shin** [comment]
Moved other subscriptions as well and removed `_initialized`

File: src/lib/slider/slider.ts [hunk 1]
```diff
@@ -410,13 +410,25 @@ export class MdSlider extends _MdSliderMixinBase
               private _changeDetectorRef: ChangeDetectorRef,
               @Optional() private _dir: Directionality) {
     super(renderer, elementRef);
+  }
+
+  ngOnInit() {
```
Comment (mmalerba) [inline src/lib/slider/slider.ts]: class should implement `OnInit`

File: src/lib/slider/slider.ts [hunk 2]
```diff
@@ -410,13 +410,25 @@ export class MdSlider extends _MdSliderMixinBase
...
     this._focusOriginMonitor
-        .monitor(this._elementRef.nativeElement, renderer, true)
-        .subscribe((origin: FocusOrigin) => this._isActive = !!origin && origin !== 'keyboard');
+        .monitor(this._elementRef.nativeElement, this._renderer, true)
+        .subscribe((origin: FocusOrigin) => {
+      this._isActive = !!origin && origin !== 'keyboard';
```
Comment (mmalerba) [inline src/lib/slider/slider.ts]: nit: indent like this:

```
this...
    .monitor(...)
    .subscribe(...
      ...
    })
```

**Comment by g1shin** [comment]
Comments have been addressed :+1:

**Pull request merged**